modules/s3-bucket/main.tf

```diff
@@ -1,53 +1,83 @@
 locals {
   enabled = module.this.enabled
 
+  aws_partition = data.aws_partition.current.partition
+
   custom_policy_account_arns = [
     for acct in var.custom_policy_account_names :
-    format("arn:%s:iam::%s:root", data.aws_partition.current.partition, module.account_map.outputs.full_account_map[acct])
+    format("arn:%s:iam::%s:root", local.aws_partition, module.account_map.outputs.full_account_map[acct])
   ]
 
-  bucket_policy = var.custom_policy_enabled ? data.aws_iam_policy_document.custom_policy[0].json : var.policy
+  bucket_policy = var.custom_policy_enabled ? data.aws_iam_policy_document.custom_policy[0].json : data.template_file.bucket_policy.rendered
 }
 
 data "aws_partition" "current" {}
```

Comment:
This should not be necessary. If we are requiring `account_map` anyway, we should use `account_map/modules/roles_to_principals` to map from account names to principal Role ARNs. (`roles_to_principals` should take `partition` as a variable with default `aws`.)